execution/src/execution.rs

```diff
@@ -179,6 +179,21 @@ impl<R: ExecutionRpc> ExecutionClient<R> {
         })
     }
 
+    pub async fn get_transaction_by_block_hash_and_index(
+        &self,
+        payload: &ExecutionPayload,
+        index: usize,
+    ) -> Result<Option<Transaction>> {
+        let tx = payload.transactions[index].clone();
+        let tx_hash = H256::from_slice(&keccak256(tx));
+        let mut payloads = BTreeMap::new();
+        payloads.insert(payload.block_number, payload.clone());
+        let tx_option = self.get_transaction(&tx_hash, &payloads).await?;
```

Comment:
Here I think we should use the rpc method you built: self.rpc.get_transaction_by_block_hash_and_index rather than self.get_transaction

Reply:
Ok! I was thinking of which one to use and chose this one(rpc level or execution level) but will implement the Rpc level implementation then

Reply:
Just make sure if you do choose to use the rpc, you need to be very careful to verify that the returned tx has the same hash as the hash at that location in the payload. Otherwise it would be possible for the rpc to send a fake tx back.

Reply:
I see, should we keep the current implementation then? As from what I understood if we use the rpc call, we will basically need to access "almost" the same data we are accessing now and then compare

Reply:
Or would be nice to have the comparison(fetching from both payload and rpc) as it will be an extra layer of security?

Reply:
I think this is good. Sorry for the late response here, missed the notification.